rust/connlib/snownet/src/node.rs

```diff
@@ -829,13 +837,15 @@ where
                 continue;
             };
 
-            if !self.allocations.contains_key(server) {
-                tracing::debug!(address = %server, "Adding new TURN server");
+            let existing = self.allocations.insert(
+                *server,
+                Allocation::new(*server, username, password.clone(), realm, self.last_now),
+            );
 
-                self.allocations.insert(
-                    *server,
-                    Allocation::new(*server, username, password.clone(), realm, self.last_now),
-                );
+            if existing.is_some() {
+                tracing::info!(address = %server, "Replaced existing allocation because credentials to TURN server changed");
+            } else {
+                tracing::info!(address = %server, "Added new TURN server");
             }
```

Comment:
I started a discussion with the str0m maintainers on whether we should call `invalidate_candidate` here on all agents and perhaps trigger an ICE restart. it could be that we are just removing the nominated candidate pair and thus need to trigger an ICE restart.

Discussion is here: https://str0m.zulipchat.com/#narrow/stream/377845-general/topic/invalidating-candidates

Reply:
We don't have to block this PR on that though. If a relay restarts, the connections won't work anymore anyway so this is just an optimisation to perhaps rescue some of the connections through an ICE restart instead of having to fail the connection via the tunnel.